### PR TITLE
:sparkles: feat: Implement payment

### PR DESCRIPTION
**Pull request by Jhonatanmizu**

### CODE REVIEW

File: src/app/provider/checkout/page.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { CookieService } from "@/app/auth/services/cookie.service";
+import { redirect } from "next/navigation";
+import { createCheckoutSession, getProductPriceSelected } from "./actions";
+import Payment from "./components/Payment";
```
Comment (Jhonatanmizu) [inline src/app/provider/checkout/page.tsx:4]:
```suggestion
import { redirect } from "next/navigation";
// Services
import { CookieService } from "@/app/auth/services/cookie.service";
// Actions
import { createCheckoutSession, getProductPriceSelected } from "./actions";
// Components
import Payment from "./components/Payment";
```

File: src/app/provider/offer/page.tsx [hunk 2]
```diff
@@ -1,16 +1,28 @@
 // Controllers
 import { getServices } from "@/app/common/controllers/service.controller";
 // Components
-import ListServices from "./components/ListServices";
 import { CookieService } from "@/app/auth/services/cookie.service";
+import { getProviderByUserId } from "@/app/common/controllers/provider.controller";
+import { redirect } from "next/navigation";
+import { getProductByPriceById } from "../profile/actions";
+import ListServices from "./components/ListServices";
```
Comment (Jhonatanmizu) [inline src/app/provider/offer/page.tsx:8]:
```suggestion
import { redirect } from "next/navigation";
// Services
import { CookieService } from "@/app/auth/services/cookie.service";
// Controllers
import { getProviderByUserId } from "@/app/common/controllers/provider.controller";
// Actions
import { getProductByPriceById } from "../profile/actions";
// Components
import ListServices from "./components/ListServices";
```

File: src/app/provider/plans/components/ProductCard.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  Button,
+  Card,
+  CardBody,
+  CardFooter,
+  CardHeader,
+  Image,
+} from "@nextui-org/react";
+import { useRouter } from "next/navigation";
+import Stripe from "stripe";
+import { setProductPriceSelected } from "../actions/setProductSelected";
```
Comment (Jhonatanmizu) [inline src/app/provider/plans/components/ProductCard.tsx:13]:
```suggestion
import { useRouter } from "next/navigation";
import Stripe from "stripe";
// Assets
import defaultImage from "@/assets/defaultplaceholder.png";
// Components
import { DynamicTypePrice } from "@/app/common/components/DynamicTypePrice";
import {
  Button,
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  Image,
} from "@nextui-org/react";
// Actions
import { setProductPriceSelected } from "../actions/setProductSelected";
```

File: src/app/provider/plans/components/ProductList.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+"use client";
+import { removeWrapper } from "@/app/common/utils";
+import { useEffect } from "react";
+import Stripe from "stripe";
+import ProductCard from "./ProductCard";
```
Comment (Jhonatanmizu) [inline src/app/provider/plans/components/ProductList.tsx:5]:
```suggestion
"use client";
import { removeWrapper } from "@/app/common/utils";
import { useEffect } from "react";
import Stripe from "stripe";
// Components
import ProductCard from "./ProductCard";
```

File: src/app/provider/plans/page.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { getProducts } from "./actions/getProducts";
+import ProductList from "./components/ProductList";
```
Comment (Jhonatanmizu) [inline src/app/provider/plans/page.tsx:2]:
```suggestion
// Actions
import { getProducts } from "./actions/getProducts";
// Components
import ProductList from "./components/ProductList";

```

**Review by Jhonatanmizu — COMMENTED** [review]
read the comments

**Review by EverNeverDev — APPROVED** [review]

**Pull request merged**